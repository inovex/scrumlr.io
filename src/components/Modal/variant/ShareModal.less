@import '../../../constants/style';

.share-modal {
  max-width: 320px;

  display: flex;
  flex-direction: column;
}

.share-modal__headline {
  align-self: flex-start;
  margin-top: @default-margin;
}

.share-modal__qr:focus {
  outline: 2px solid @color-mint;
}

.share-modal__invite-button {
  border: 0;
  background: none;
  outline: none;
  width: 100%;

  font-size: @theme-font-size;
  font-weight: bold;
  text-transform: uppercase;

  padding: @default-padding @large-padding;

  margin-top: @default-margin;
  margin-bottom: @default-margin;
  color: @color-white;
  background: @color-mint;

  transition: background .2s ease-in-out, box-shadow .2s ease-in-out;
}

.share-modal__invite-button:focus,
.share-modal__invite-button:hover {
  background: darken(@color-mint, 15%);
  box-shadow: 0 @base-unit*2 @base-unit*4 0 fade(@color-dark, 15%);
}

.share-modal__copy-text {
  font-size: @theme-font-size - @base-unit / 2;
  color: @color-alternative-gray;
}

.share-modal__copy-text--hidden {
  visibility: hidden;
}