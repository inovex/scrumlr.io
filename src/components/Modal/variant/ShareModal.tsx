import * as cx from 'classnames';
import * as React from 'react';
import * as CopyToClipboard from 'react-copy-to-clipboard';

import './ShareModal.css';
import Modal from '../Modal';

export interface ShareModalProps {
  onClose: () => void;
}

export interface ShareModalState {
  showCopyInfo: boolean;
}

export class ShareModal extends React.Component<
  ShareModalProps,
  ShareModalState
> {
  constructor(props: ShareModalProps) {
    super(props);
    this.state = { showCopyInfo: false };
  }

  displayCopyMessage = () => {
    this.setState({ ...this.state, showCopyInfo: true });
  };

  render() {
    const { onClose } = this.props;

    const link = window.location.href.replace('board', 'join');
    const encoded = encodeURI(link).replace('#', '%23');

    return (
      <Modal onClose={onClose} onSubmit={onClose}>
        <div className="share-modal">
          <h2 className="share-modal__headline">Invite</h2>

          <a href={link} className="share-modal__qr">
            <img
              src={`https://chart.googleapis.com/chart?cht=qr&chs=400x400&chld=L|0&chl=${encoded}`}
              className="share-modal__qr-image"
            />
          </a>
          <CopyToClipboard text={link} onCopy={this.displayCopyMessage}>
            <button type="button" className="share-modal__invite-button">
              Copy invite URL
            </button>
          </CopyToClipboard>
          <span
            className={cx('share-modal__copy-text', {
              'share-modal__copy-text--hidden': !this.state.showCopyInfo
            })}
          >
            URL copied to clipboard
          </span>
        </div>
      </Modal>
    );
  }
}

export default ShareModal;
