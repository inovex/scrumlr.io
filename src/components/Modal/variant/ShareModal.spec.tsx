import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { ShareModal, ShareModalProps, ShareModalState } from './ShareModal';

jest.mock('../Modal');

describe('<ShareModal />', () => {
  let closeFunction: () => void;
  let shallowWrapper: ShallowWrapper<ShareModalProps, ShareModalState>;

  const baseUrl = 'http://localhost:3000/#';
  const boardId = '-L7ucuDefYZL6uvh-Q4C';

  beforeEach(() => {
    closeFunction = jest.fn();
    shallowWrapper = shallow(<ShareModal onClose={closeFunction} />);

    Object.defineProperty(window.location, 'href', {
      writable: true,
      value: `${baseUrl}/board/${boardId}`
    });
  });

  it('should match the expected structure', () => {
    expect(shallowWrapper.html()).toMatchSnapshot();
  });

  it('should provide the correct sharing url', () => {
    expect(
      shallowWrapper
        .find('a')
        .at(0)
        .props().href
    ).toEqual(`${baseUrl}/join/${boardId}`);
  });

  it('should hide the copy to clipboard message initially', () => {
    expect(shallowWrapper.find('.share-modal__copy-text--hidden')).toHaveLength(
      1
    );
  });
});
