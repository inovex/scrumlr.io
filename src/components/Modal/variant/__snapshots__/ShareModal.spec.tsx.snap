// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShareModal /> should match the expected structure 1`] = `"<div><div class=\\"share-modal\\"><h2 class=\\"share-modal__headline\\">Invite</h2><a href=\\"http://localhost/\\" class=\\"share-modal__qr\\"><img src=\\"https://chart.googleapis.com/chart?cht=qr&amp;chs=250x250&amp;chld=L|0&amp;chl=http://localhost/\\"/></a><button type=\\"button\\" class=\\"share-modal__invite-button\\">Copy invite URL</button><span class=\\"share-modal__copy-text share-modal__copy-text--hidden\\">URL copied to clipboard</span></div></div>"`;
